### PR TITLE
fix Sprockets3 compatibility

### PR DESCRIPTION
**Pull request by sevaorlov**
It fixed the problem for our app, but will be good if someone else can try it one more time.

Continuation of https://github.com/railsware/js-routes/pull/170

### CODE REVIEW

File: lib/js_routes/engine.rb [hunk 1]
```diff
@@ -4,15 +4,27 @@ class JsRoutes
   SPROCKETS3 = Gem::Version.new(Sprockets::VERSION) >= Gem::Version.new('3.0.0')
   class Engine < ::Rails::Engine
 
-    initializer 'js-routes.dependent_on_routes', after: "sprockets.environment" do
+    if SPROCKETS3
+      initializer 'js-routes.dependent_on_routes', after: :engines_blank_point, before: :finisher_hook do
```
Comment (sevaorlov) [inline lib/js_routes/engine.rb]: https://github.com/fnando/i18n-js/pull/371/files#r45864633

**Comment by le0pard** [comment]
@sevaorlov looks like tests is broken.

**Comment by sevaorlov** [comment]
@le0pard, yep, I've added a commit to fix them

File: lib/js_routes/engine.rb (hunk 1 above)
Comment (bogdan) [inline lib/js_routes/engine.rb]: Why do you combo `after` and `before` options at the same time? What does it mean?

Comment (bogdan) [inline lib/js_routes/engine.rb]: @sevaorlov any response here?

Comment (sevaorlov) [inline lib/js_routes/engine.rb]: @bogdan, we need `after: :engines_blank_point` in order to run this block after engines are initialized, `finisher_hook` is used as border as before this hook all initializations should be completed http://guides.rubyonrails.org/configuring.html#initializers

**Pull request merged**